apis/v1beta1/validation/httproute.go

```diff
@@ -287,3 +289,39 @@ func hasExactlyOnePrefixMatch(matches []gatewayv1b1.HTTPRouteMatch) bool {
 
 	return true
 }
+
+// validateHTTPParentRefs validates that if ParentRefs includes 2 or more references
+// to the same parent (based on kind, name, and namespace), those ParentRefs must
+// specify unique SectionName values.
+func validateHTTPParentRefs(parentRefs []gatewayv1b1.ParentReference, path *field.Path) field.ErrorList {
+	var errs field.ErrorList
+	if len(parentRefs) <= 1 {
+		return nil
+	}
+	type sameKindParentRefs struct {
+		name      gatewayv1b1.ObjectName
+		namespace gatewayv1b1.Namespace
+		kind      gatewayv1b1.Kind
+	}
+	m := make(map[sameKindParentRefs][]gatewayv1b1.SectionName)
+	for i, p := range parentRefs {
+		if p.Namespace == nil {
```

Comment:
Since we need to validate the ParentRefs, I'd prefer not to touch the original resource but instead, to rely only on additional variables; it'd look cleaner and easier to understand. WDYT?

Reply:
Do you mean that we only check the addition/incremental ParentRefs instead of checking all ParentRefs? Can you explain more about the reason? And IMO, there is no way to only check incremental resource in Kubernetes webhook validation, because it cannot tell the incremental or original resource, correct me if I'm wrong.

Reply:
my point is only related to the way this code is written: you are initializing the nil fields in the `parentRefs`. I'm just suggesting rewriting this portion of the code to leave the original `parentRefs` untouched and rely only on the additional data structure. E.g.,
```go
t := sameKindParentRefs{name: p.Name, namespace: *p.Namespace, kind: *p.Kind}
if t.namespace == nil {
    t.namespace = new(gatewayv1b1.Namespace)
}
...
```